packages/core/src/external/commonwell/document/__tests__/document-downloader-local.test.ts

```diff
@@ -0,0 +1,135 @@
+        size: 0,
+        contentType: faker.system.mimeType(),
+        createdAt: undefined,
+      });
+    });
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+    afterAll(() => {
+      jest.restoreAllMocks();
+    });
+
+    const queryMeta = organizationQueryMeta(faker.company.name(), {
+      npi: faker.string.alpha(10),
+    });
+    const downloader = new DocumentDownloaderLocal({
+      region: "us-east-1",
+      bucketName: faker.lorem.word(),
+      commonWell: {
+        api: {} as CommonWellAPI,
+        queryMeta,
+      },
+    });
+    const downloadedFile = {
+      bucket: faker.lorem.word(),
+      key: faker.lorem.sentence(),
+      location: faker.internet.url(),
+      size: faker.number.int(),
+      contentType: faker.system.mimeType(),
+    };
+    const requestedFileInfo: FileInfo = {
+      name: faker.lorem.word(),
+      location: faker.internet.url(),
+    };
+
+    it("parses b64 encoded txt and returns txt content type", async () => {
+      const res = await downloader.parseXmlFile({
+        contents: getCdaWithB64EncodedText(),
+        ...downloadedFile,
+        requestedFileInfo,
+      });
+      expect(res).toBeTruthy();
+      expect(s3Upload_mock).toHaveBeenCalledWith(
+        expect.objectContaining({ ContentType: TXT_MIME_TYPE })
+      );
+    });
+
+    it("parses b64 encoded tiff and returns tiff content type", async () => {
+      const res = await downloader.parseXmlFile({
+        contents: getCdaWithB64EncodedTiff(),
+        ...downloadedFile,
+        requestedFileInfo,
+      });
+      expect(res).toBeTruthy();
+      expect(s3Upload_mock).toHaveBeenCalledWith(
+        expect.objectContaining({ ContentType: TIFF_MIME_TYPE })
+      );
+    });
+
+    it("parses b64 encoded pdf and returns pdf content type", async () => {
+      const res = await downloader.parseXmlFile({
+        contents: getCdaWithB64EncodedPdf(),
+        ...downloadedFile,
+        requestedFileInfo,
+      });
+      expect(res).toBeTruthy();
+      expect(s3Upload_mock).toHaveBeenCalledWith(
+        expect.objectContaining({ ContentType: PDF_MIME_TYPE })
+      );
+    });
+
+    it("parses b64 encoded png and returns png content type", async () => {
+      const res = await downloader.parseXmlFile({
+        contents: getCdaWithB64EncodedPng(),
+        ...downloadedFile,
+        requestedFileInfo,
+      });
+      expect(res).toBeTruthy();
+      expect(s3Upload_mock).toHaveBeenCalledWith(
+        expect.objectContaining({ ContentType: PNG_MIME_TYPE })
+      );
+    });
+
+    it("parses b64 encoded jpeg and returns jpeg content type", async () => {
+      const res = await downloader.parseXmlFile({
+        contents: getCdaWithB64EncodedJpeg(),
+        ...downloadedFile,
+        requestedFileInfo,
+      });
+      expect(res).toBeTruthy();
+      expect(s3Upload_mock).toHaveBeenCalledWith(
+        expect.objectContaining({ ContentType: JPEG_MIME_TYPE })
+      );
+    });
```

Comment:
we need a unit test here for the octet binary stream type too. Would probably have caught the bug in the logic for text files since the binary file would 50/50 get classified as a text with only 6 bytes beinge evaluated